### PR TITLE
Removing hard-coded shard sizes from the Dataflow tests 

### PR DESCRIPTION
**Pull request by davidadamsphd**

### CODE REVIEW

File: src/test/java/org/broadinstitute/hellbender/engine/dataflow/datasources/VariantShardUnitTest.java [hunk 1]
```diff
@@ -30,4 +30,10 @@ public void getVariantShardsFromIntervalTest(GATKRead read, Iterable<VariantShar
             List<VariantShard> foundShards = VariantShard.getVariantShardsFromInterval(read);
             Assert.assertEquals(foundShards, expectedShards);
     }
+
+    @Test
+    public void verifyDivisibilityWithRefShard() {
```
Comment (lbergelson) [inline src/test/java/org/broadinstitute/hellbender/engine/dataflow/datasources/VariantShardUnitTest.java]: This test makes sense but it seems funny having it here.  Maybe it should just move into the test data class itself?

File: src/test/java/org/broadinstitute/hellbender/engine/dataflow/ReadsPreprocessingPipelineTestData.java [hunk 2]
```diff
@@ -47,7 +47,10 @@
      * @param clazz The class to be used to back the GATKRead, either Read.class, or SAMRecord.class.
      */
     public ReadsPreprocessingPipelineTestData(Class<?> clazz) {
-        readStartLength = Arrays.asList(KV.of(100, 50), KV.of(140, 100), KV.of(1000000, 10), KV.of(2999999, 10));
+        final int shardRatio = ReferenceShard.REFERENCE_SHARD_SIZE / VariantShard.VARIANT_SHARDSIZE;
+        readStartLength = Arrays.asList(KV.of(100, 50), KV.of(140, 100),
+                KV.of(ReferenceShard.REFERENCE_SHARD_SIZE, 10),
+                KV.of(3*ReferenceShard.REFERENCE_SHARD_SIZE - 1, 10));
 
         // TODO Make reads construction more general (issue #687).
```
Comment (lbergelson) [inline src/test/java/org/broadinstitute/hellbender/engine/dataflow/ReadsPreprocessingPipelineTestData.java]: I think this comment is resolved with this pull request?

**Comment by lbergelson** [comment]
@davidadamsphd Looks good. Just two very minor comments.  I didn't actually tweak the shard size and run it to test, did you try it?  

Feel free to merge once you're ready.

File: src/test/java/org/broadinstitute/hellbender/engine/dataflow/datasources/VariantShardUnitTest.java (hunk 1 above)
Comment (davidadamsphd) [inline src/test/java/org/broadinstitute/hellbender/engine/dataflow/datasources/VariantShardUnitTest.java]: Moved it.

File: src/test/java/org/broadinstitute/hellbender/engine/dataflow/ReadsPreprocessingPipelineTestData.java (hunk 2 above)
Comment (davidadamsphd) [inline src/test/java/org/broadinstitute/hellbender/engine/dataflow/ReadsPreprocessingPipelineTestData.java]: Yep, removed.

**Comment by davidadamsphd** [comment]
@lbergelson, I tried 300,000 for the ref shard and it worked, I tried 300,001 and it didn't work (as expected).

**Comment by lbergelson** [comment]
:+1:

**Pull request merged**